platform-emf/src/main/java/com/softicar/platform/emf/data/table/EmfDataTablePath.java

```diff
@@ -0,0 +1,132 @@
+package com.softicar.platform.emf.data.table;
+
+import com.softicar.platform.common.core.i18n.IDisplayString;
+import com.softicar.platform.common.core.utils.CastUtils;
+import com.softicar.platform.common.string.Imploder;
+import com.softicar.platform.common.string.hash.Hash;
+import com.softicar.platform.dom.elements.popup.DomPopup;
+import com.softicar.platform.dom.elements.popup.compositor.CurrentDomPopupCompositor;
+import com.softicar.platform.dom.elements.tab.DomTabBar;
+import com.softicar.platform.dom.node.IDomNode;
+import com.softicar.platform.emf.form.popup.EmfFormPopup;
+import com.softicar.platform.emf.page.IEmfPageContentElement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.Optional;
+
+public class EmfDataTablePath {
+
+	private final String pathString;
+
+	public EmfDataTablePath(IEmfDataTable<?> table) {
+
+		Objects.requireNonNull(table);
+
+		var path = new PathList();
+		IDomNode node = table;
+		do {
+			var n = node;
```

Comment:
what's the purpose of the `n`? saving 3 letter per line? ;)

Reply:
![image](https://user-images.githubusercontent.com/15086658/180245261-7aef0a27-bc0d-45af-afbf-374bfb5fc7e9.png)
...that's the point.

Reply:
i see the point :)